### PR TITLE
SRCH-4369 Upgrade to Rails 7.

### PR DESCRIPTION
**Pull request by luisgmetzger**
## Summary
- This PR upgrades the Rails gem to 7 from 6. 
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
 
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

**Comment by luisgmetzger** [comment]
@lorawoodford  - I commented out the code climate report temporarily. I need to get access to the code climate report ID as code climate is complaining about the following: 

```
Error: you must supply a CC_TEST_REPORTER_ID ENV variable or pass it via the -r flag
Usage:
  cc-test-reporter after-build [flags]

Flags:
  -s, --batch-size int               batch size for source files (default 500)
  -e, --coverage-endpoint string     endpoint to upload coverage information to (default "https://api.codeclimate.com/v1/test_reports")
  -t, --coverage-input-type string   type of input source to use [clover, cobertura, coverage.py, excoveralls, gcov, gocov, jacoco, lcov, lcov-json, simplecov, xccov, dotcover]
      --exit-code int                exit code of the test run
  -r, --id string                    reporter identifier
      --insecure                     send coverage insecurely (without HTTPS)
  -p, --prefix string                the root directory where the coverage analysis was performed (default "/home/circleci/app")
  ```
  
If everything looks good, I will get this ID and add it in on Monday.

**Comment by lorawoodford** [comment]
@luisgmetzger Can I recommend you reassign this review to @stevenbarragan and the two of you touch base on the plan here?  Looking at #231 I want to make sure you aren't working at cross streams.

Regarding the CodeClimate issue, have you followed the in the Confluence doc "How to run Code Climate on your repo"?   I can ping you a link if you need it.  We definitely can't approve this until we've got working code coverage reporting.

File: app/models/concerns/aliased_index.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ module AliasedIndex
 
   module ClassMethods
     def timestamped_index_name
-      [base_name, Time.current.to_s(:number)].join('-')
+      [base_name, Time.current.strftime('%Y%m%d%H%M%S%L')].join('-')
```
Comment (stevenbarragan) [inline app/models/concerns/aliased_index.rb:12]: `Time.current.to_s(:number)` still works on rails 7. I don't think we need to upgrade anything in here

File: config/application.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 require "rails"
 # Pick the frameworks you want:
-# require "active_model/railtie"
+require "active_model/railtie"
```
Comment (stevenbarragan) [inline config/application.rb:7]: Do we really need to require this now?

File: Gemfile [hunk 3]
```diff
@@ -22,6 +22,8 @@ gem 'elasticsearch-transport', '~> 6.0'
 gem 'flickraw', '~> 0.9.8'
 gem 'hashie', '~> 3.5.7'
 gem 'mock_redis', '~> 0.17.3'
+# Keeping a version of nokogiri that supports both Rails 6 and 7
+gem 'nokogiri', '~> 1.15.3'
```
Comment (stevenbarragan) [inline Gemfile:26]: We don't need to lock nokigiri's version

**Review by stevenbarragan — CHANGES_REQUESTED** [review]
One change and we should good to go

File: Gemfile (hunk 3 above)
Comment (luisgmetzger) [reply Gemfile:26]: @stevenbarragan  - Ah yes, forgot about this one. I forgot to remove this as I no longer needed this.

Comment (luisgmetzger) [reply Gemfile]: @stevenbarragan  - I remembered why I did this. In circleci, we have a matrix for the ruby version of 2.7.5 where we have a nokogiri depedency error during bundle install step: 

```
The following gems are missing
 * nokogiri (1.16.0)
 * mini_portile2 (2.8.5)
Install missing gems with `bundle install`
Fetching gem metadata from https://rubygems.org/.........
nokogiri-1.16.0 requires ruby version >= 3.0.0, which is incompatible with the
current version, 2.7.5
```

I am thinking we either state the explicit older version of nokogiri, `1.15.3` or we can remove the ruby 2.7.5 version from the matrix in circleci, thoughts?

Comment (stevenbarragan) [reply Gemfile]: Let's get rid of the ruby 2.7.5 and the 3.0.5 version checks instead. We don't need them anymore.

**Review by stevenbarragan — APPROVED** [review]
Ideally we should only keep the version we use in the server

**Pull request merged**